### ISSUE TITLE
Sometimes the modal blinks on close

### ISSUE THREAD

**Issue by zubko** [zhantx/react-native-flex-modal#1]
This library looks really promising.
But I can't use it because in about 50% of cases it blinks on close animation.
I'm using long slide path (no scale)
```
        positionYIn={400}
        positionYOut={400}
```
and in my test the view is quite big
```
<View
          style={{
            height: 400,
            width: 320,
            backgroundColor: 'white',
            alignItems: 'center',
            justifyContent: 'center',
          }}
```
I can't show a video from a client's app, I will make some test project if I have time later.

**Comment by zubko** [comment]
I was able to fix that by having 2 state flags for visibility `isOpen` and `isModalOpen`,
and using them like this
```
<Modal
        animationType={'none'}
        transparent={true}
        visible={this.state.isModalOpen}
        onRequestClose={this.backPressed}
      >
        {this.state.isOpen
          ? <Animated.View
```
then in `close()` I've dropped first `isOpen` and then `isModalOpen`:
```
this.setState(
        {
          isOpen: false,
          ...
        },
        () => {
          this.setState({
            isModalOpen: false,
          });
        }
);
```
And in `open()` I set them both together.
Looks like `<Modal>` needs 1 render being empty before its `visible` will be `false`, otherwise when it hides it may blink its contents. 
To see this issue every time, change `Modal` animation from `none` to `slide`. The slide on hide should not be visible (because the contents is hidden after animation). But it will slide down with visible contents.
At least in my case. I'm using RN 0.44.
